src/plugins/graph/adornments/plotted-function/plotted-function-adornment-model.ts

```diff
@@ -0,0 +1,49 @@
+import { Instance, types } from "mobx-state-tree";
+import { AdornmentModel, IAdornmentModel } from "../adornment-models";
+import { kPlottedFunctionType, kPlottedFunctionValueTitleKey, FormulaFn } from "./plotted-function-adornment-types";
+
+export const PlottedFunctionInstance = types.model("PlottedFunctionInstance", {})
+  .volatile(self => ({
+    // This is being hard coded to x**2 for now.
+    // It should be reverted to NaN once we've set up correct functions.
+    // formulaFunction: (x: number) => NaN,
+    formulaFunction: (x: number) => x**2,
+  }))
+  .actions(self => ({
+    setValue(formulaFunction: FormulaFn) {
+      self.formulaFunction = formulaFunction;
+    }
+  }));
+
+export const PlottedFunctionAdornmentModel = AdornmentModel
+  .named("PlottedFunctionAdornmentModel")
+  .props({
+    type: types.optional(types.literal(kPlottedFunctionType), kPlottedFunctionType),
+    labelTitle: types.optional(types.literal(kPlottedFunctionValueTitleKey), kPlottedFunctionValueTitleKey),
+    plottedFunctions: types.map(PlottedFunctionInstance),
+    error: ""
+  })
+  .actions(self => ({
+    setError(error: string) {
+      self.error = error;
+    },
+    addPlottedFunction(formulaFunction: FormulaFn, key="{}") {
```

Comment:
I'm not sure why the default key is "{}".  It is leading to IDs in the SVG like `plotted-function-` .  Or maybe the fix is to override this default when calling this.

Reply:
🤷 I changed it to "default-function-key".

Reply:
Turns out the key is generated by calling `toJSON` on an object, so the "{}" default key is for empty objects. My attempt to change it to something less confusing didn't work, so I'm just going to keep it as is for now.